### ISSUE TITLE
java.lang.IllegalStateException: Variable substitution depth too large: 20

### ISSUE THREAD

**Issue by azalio** [MetaCenterCloudPuppet/cesnet-hadoop#7]
Hello!
I have 36 data dirs on the one server.
Then I start data node I get this error:

```
2018-06-22 15:11:47,015 INFO org.apache.hadoop.hdfs.server.datanode.DataNode: registered UNIX signal handlers for [TERM, HUP, INT]
2018-06-22 15:11:47,157 FATAL org.apache.hadoop.hdfs.server.datanode.DataNode: Exception in secureMain
java.lang.IllegalStateException: Variable substitution depth too large: 20 file:///data/1//${user.name}/dfs/data,file:///data/2//${user.name}/dfs/data,file:///data/3//${user.name}/dfs/data,file:///data/4//${user.name}/dfs/data,file:///data/5//${user.name}/dfs/data,file:///data/6//${user.name}/dfs/data,file:///data/7//${user.name}/dfs/data,file:///data/8//${user.name}/dfs/data,file:///data/9//${user.name}/dfs/data,file:///data/10//${user.name}/dfs/data,file:///data/11//${user.name}/dfs/data,file:///data/12//${user.name}/dfs/data,file:///data/13//${user.name}/dfs/data,file:///data/14//${user.name}/dfs/data,file:///data/15//${user.name}/dfs/data,file:///data/16//${user.name}/dfs/data,file:///data/17//${user.name}/dfs/data,file:///data/18//${user.name}/dfs/data,file:///data/19//${user.name}/dfs/data,file:///data/20//${user.name}/dfs/data,file:///data/21//${user.name}/dfs/data,file:///data/22//${user.name}/dfs/data,file:///data/23//${user.name}/dfs/data,file:///data/24//${user.name}/dfs/data,file:///data/25//${user.name}/dfs/data,file:///data/26//${user.name}/dfs/data,file:///data/27//${user.name}/dfs/data,file:///data/28//${user.name}/dfs/data,file:///data/29//${user.name}/dfs/data,file:///data/30//${user.name}/dfs/data,file:///data/31//${user.name}/dfs/data,file:///data/32//${user.name}/dfs/data,file:///data/33//${user.name}/dfs/data,file:///data/34//${user.name}/dfs/data,file:///data/35//${user.name}/dfs/data,file:///data/36//${user.name}/dfs/data
        at org.apache.hadoop.conf.Configuration.substituteVars(Configuration.java:1058)
        at org.apache.hadoop.conf.Configuration.get(Configuration.java:1078)
        at org.apache.hadoop.conf.Configuration.getTrimmedStringCollection(Configuration.java:1974)
        at org.apache.hadoop.hdfs.server.datanode.DataNode.getStorageLocations(DataNode.java:2473)
        at org.apache.hadoop.hdfs.server.datanode.DataNode.instantiateDataNode(DataNode.java:2465)
        at org.apache.hadoop.hdfs.server.datanode.DataNode.createDataNode(DataNode.java:2516)
        at org.apache.hadoop.hdfs.server.datanode.DataNode.secureMain(DataNode.java:2698)
        at org.apache.hadoop.hdfs.server.datanode.DataNode.main(DataNode.java:2722)
2018-06-22 15:11:47,160 INFO org.apache.hadoop.util.ExitUtil: Exiting with status 1
2018-06-22 15:11:47,161 INFO org.apache.hadoop.hdfs.server.datanode.DataNode: SHUTDOWN_MSG:
```

For fix this error I replaced `${user.name}` with actual hdfs username.
For now, I just created variable 

```
class hadoop::params {

  $hadoop_user_name = lookup('hadoop::params:hadoop_user_name', String, 'first', '${user.name}')
```
and use it in code:
```
  $hdfs_datanode_suffix = "${::osfamily}-${::operatingsystem}" ? {
    /RedHat-Fedora/ => "/$hadoop_user_name/dfs/datanode",
    /Debian|RedHat/ => "/$hadoop_user_name/dfs/data",
  }
```
But variable `${user.name}` widely used and I don't know how to fix it in the right way.

**Comment by valtri** [comment]
Apache default is `file://${hadoop.tmp.dir}/dfs/data` (which uses the `${user.name}`), Cloudera packaging has `file:///var/lib/hadoop-hdfs/cache/hdfs/dfs/name` in initial empty configuration.

I guess using `${user.name}` has probably not much sense here and the suffix can be safely hardcoded to "hdfs".

**Comment by valtri** [comment]
Fix released in cesnet-hadoop 2.5.0.
